[Javier Lim] iP

ijavierja opened the pull request.

Commit:
Add Gradle support

ijavierja committed:
added new greeting, Duke can now echo, Duke can now exit

ijavierja committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ijavierja committed:
level 2: add and list operation added

ijavierja committed:
mark as done. Incomplete todo, deadline and event operations

ijavierja committed:
completed todo, event, deadline operation

ijavierja committed:
placed Task and TaskType into Duke class for automated testing

ijavierja committed:
error handling added

ijavierja committed:
done is updated. delete method is added. Main method cleaned up.

ijavierja committed:
Remove incorrect notes

Incomplete note on some methods were removed

ijavierja committed:
minor edit

ijavierja committed:
Merge remote-tracking branch 'origin/master'

ijavierja committed:
Merge branch 'master' of https://github.com/ijavierja/ip

* 'master' of https://github.com/ijavierja/ip:
  Remove incorrect notes

ijavierja committed:
Add save method

Save method puts the list of tasks into a text file.
This allows the client to access tasks even if Duke has been closed.
The path of the text file can be found in the Duke.java file as a global variable at the top of the file.

ijavierja committed:
Put classes into package duke

ijavierja committed:
Add LocalDateTime attribute to Task class & modified deadline method to allow only a date time description

Level 8. Date and Times.
The Task class was modified to include a new field of type LocalDateTime.
This is to keep the Time and Date for the tasks.
Deadline method was also modified to only allow for a date and time description.

ijavierja committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

ijavierja committed:
Add Ui class

ui class deals with interaction with the user. Ui class takes commands from the user as well as print output to user.

ijavierja committed:
Correct add Ui class

ijavierja committed:
Add TaskList class

TaskList stores an ArrayList<Task> and deals with all changes to this list. Additionally, the class deals with all outputs needed from this arraytist

ijavierja committed:
Add DukeException and Storage classes

Storage class deals with loading tasks from the file and saving tasks in the file
DukeException is thrown when no file is present with the name represented by filePath

ijavierja committed:
Add Command and Parser classes

Command is an interface with two abstract methods.
Parser deals with making sense of the user command. A Command instance is implemented using Parser's static parse method.

ijavierja committed:
Add ParserTest and TaskTest classes; Modified Command interface's method, execute, output type to String from void

Test classes uses JUnit for testing of the classes' methods.

ijavierja committed:
Add MANIFEST.MF file

ijavierja committed:
Add JavaDoc comments

Only header comments for classes have been added

ijavierja committed:
Tweak the code to comply with a coding standard

ijavierja committed:
Add Find method

List all task that contains the phrase

ijavierja committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

ijavierja committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/TaskType.java
#	src/main/java/Ui.java

xz0127 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,74 @@
+
+    /**
+     * Constructor which takes in file path of the storage file.
+     * @param filePath
```
Perhaps add a description to the parameter name in the Javadoc comments to make it clearer?

xz0127 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,446 @@
+	 * Returns the first word in the string indicated by the first blank space.
+	 * If there are no blank space, the entire string is returned.
+	 *
+	 * @param string
```
Perhaps add a description for the return value for Javadoc comments to make it more complete? I noticed this issue in several other places as well.

xz0127 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,446 @@
+
+public class Parser {
+	public static Command parse(String fullCommand) throws DukeException {
+		if(fullCommand.equals("list")) {
```
I like the way the if-else conditions are constructed. They are really clear!:) 👍

xz0127 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,446 @@
+	public static Command list() {
+		return new Command() {
+			@Override
+			public String execute(TaskList taskList, Ui ui, Storage storage) {
```
Should this part be extracted out? Perhaps consider creating multiple command class to handle different commands in order to improve the readability of following command method? :)

xz0127 reviewed (commented):
Overall, I found your code easy to read in most of the parts except that some methods could be extracted into separate classes to further improve the readability and prevent the method being too complicated, if make sense. I noticed that there are several violations of the coding standard such as the completeness of Javadoc comments and indentation.

ijavierja committed:
Add gradle support

howtoosee commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import duke.Task;
```
wrong order of import (alphabetical)

howtoosee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,71 @@
+		this.string = string;
+		this.dateTime = Optional.empty();
+	}
+	public Task(duke.TaskType taskType, boolean isDone, String string, Optional<LocalDateTime> dateTime) {
```
damn optional 0.0

howtoosee commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,44 @@
+		System.out.println(logo);
+	}
+
+	public String readCommand() {
```
missing javadocs for methods in this class

howtoosee commented on src/test/java/TaskTest.java:
```diff
@@ -0,0 +1,27 @@
+
+import static org.junit.Assert.assertEquals;
+
+public class TaskTest {
```
missing javadocs

howtoosee commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,39 @@
+
+import static org.junit.Assert.assertEquals;
+
+public class ParserTest {
```
missing javadocs

howtoosee commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,43 @@
+ * TaskList contains an ArrayList of Task.
+ * It serves as a barrier to directly using the ArrayList class so as to limit the functions available to the client.
+ */
+public class TaskList {
```
missing javadocs for methods in this classs

howtoosee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,71 @@
+	}
+	public String getDoneString() {
+		String string;
+		if(isDone){
```
missing spaces between parentheses and brace

howtoosee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,71 @@
+ */
+
+public class Task {
+	duke.TaskType taskType;
```
use an import rather than referring to `duke.TaskType`

howtoosee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,71 @@
+public class Task {
+	duke.TaskType taskType;
+	boolean isDone;
+	String string;
```
is there a better name than `string`? this isn't very informative

howtoosee commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,90 @@
+
+	public ArrayList<Task> load() throws DukeException {
+		File file = new File(filePath);
+		ArrayList<Task> list = new ArrayList<>();
```
i think my tutor mentioned that 'list' as a collection name isnt very intuitive and informative.

howtoosee reviewed (commented):
Very clean and readable code! goodjob! However, i noticed that you use 8 spaces for indentation while the convention is to use 4. also, remember to add javadocs for public methods!

other than that, code is clear and concise!

howtoosee reviewed (commented):
Very clean and readable code! goodjob! However, i noticed that you use 8 spaces for indentation while the convention is to use 4. also, remember to add javadocs for public methods!

other than that, code is clear and concise!

// edit: sorry my internet lagged and i submitted two reviews oops

ijavierja committed:
Add GUI part 4 successfully

ijavierja committed:
Add GUI

ijavierja committed:
Edit Test classes

ijavierja committed:
Add Assertion

ijavierja committed:
Improve Code Quality

Package classes into sub directories and imported relevant classes.

Duke:
Remove unused code (from previous iterations of GUI) from duke class.
Remove Constructor with single param filename.
Update main to use no param constructor.
Made changes to the JavaDocs.

Storage:
Add switch statement for Storage class

Task:
used unicode number instead of the cross and tick characters.

ijavierja committed:
Abstract Command types from Parser class

ijavierja committed:
Merge pull request #2 from ijavierja/branch-A-CodeQuality

Branch a code quality

ijavierja committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Parser.java
#	src/main/java/storage/Storage.java
#	src/main/java/task/Task.java

ijavierja committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Parser.java
#	src/main/java/storage/Storage.java
#	src/main/java/task/Task.java

ijavierja committed:
Merge branch 'branch-A-CodeQuality'

ijavierja committed:
Add Tagging

ijavierja committed:
Improve GUI

ijavierja committed:
Add Userguide

ijavierja committed:
Correct bug in Gui

ijavierja committed:
Add screenshot

ijavierja committed:
Edit the README.md file

ijavierja committed:
Change readme

ijavierja committed:
Edit readme

ijavierja committed:
A-UserGuide

ijavierja committed:
edit readme file

ijavierja committed:
no message

ijavierja committed:
remove ui.png

ijavierja committed:
add ui.png
